src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,844 @@
+                    path = os.getcwd() + '/'+ path
+                items = os.listdir(path)
+                print_list(items)    
+        except :
+            traceback.print_exc(file = sys.stdout)
+       
+    def do_lpwd(self, arglist):
+        """
+        Prints the absolute path of the current local directory
+        """
+        print(os.getcwd())
+
+    def do_df(self, arglist):
+        """
+
+        """
+        print('{:10s}\t{:5s}\t{:15s}{:10s}{}'.format("1K-blocks", "Used", "Available", "Use%", "Stored on"))
+        for i in list_items(cephfs.getcwd()):
+            if is_file_exists(i.d_name):
+                stat = cephfs.stat(i.d_name)
+                available = stat.st_blksize - stat.st_size
+                use = (stat.st_size*100 // stat.st_blksize)
+                print('{:10d}\t{:5d}\t{:10d}\t{:5s} {}'.format(stat.st_blksize, stat.st_size, available, str(int(use)) + '%', cephfs.getxattr(i.d_name, 'ceph.file.layout.pool').decode('utf-8')))
+
+
+    locate_parser = argparse.ArgumentParser(description = 'Find file within file system')
+    locate_parser.add_argument('name', help = 'name', type = str)
+    locate_parser.add_argument('-c', '--count', action = 'store_true', help = 'Count list of items located.')
+    locate_parser.add_argument('-i', '--ignorecase', action = 'store_true', help = 'Ignore case')
+
+    @with_argparser(locate_parser)
+    def do_locate(self, args):
+        """
+        Find a file within the File System
+        """
+        if args.name.count('*') == 1 :
+            if args.name[0] == '*':
+                args.name += '/'
+            elif args.name[-1] == '*':
+                args.name = '/'+ args.name
+        args.name = args.name.replace('*', '')
+        if args.ignorecase :
+            locations = [i for i in sorted(set(dirwalk(cephfs.getcwd().decode('utf-8')))) if args.name.lower() in i.lower()]
+        else :
+            locations = [i for i in sorted(set(dirwalk(cephfs.getcwd().decode('utf-8')))) if args.name in i]
+        if args.count :
+            print(len(locations))
+        else:
+            print('\n'.join(locations))
+
+    def complete_du(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    du_parser = argparse.ArgumentParser(description = 'Disk Usage of a Directory')
+    du_parser.add_argument('dirs', type = str, help = 'Name of the directory.', nargs = '*', default = [''])
+
+    @with_argparser(du_parser)
+    def do_du(self, args):
+        if args.dirs[0] == '':
+            args.dirs[0] = cephfs.getcwd().decode('utf-8')
+        for dir_ in args.dirs :
+            for i in reversed(sorted(set(dirwalk(dir_)))):
+                try :
+                    print('{:10s} {}'.format(humansize(int(cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes').decode('utf-8'))), '.' + i))
+                except :
+                    continue
```

Comment:
Try this patch:

````
@@ -802,18 +808,12 @@ class CephfsShell(Cmd):
         return self.complete_filenames(text, line, begidx, endidx)
 
     du_parser = argparse.ArgumentParser(description = 'Disk Usage of a Directory')
-    du_parser.add_argument('dirs', type = str, help = 'Name of the directory.', nargs = '*', default = [''])
+    du_parser.add_argument('dirs', type = str, help = 'Name of the directory.', nargs = '*', default = ['.'])
 
     @with_argparser(du_parser)
     def do_du(self, args):
-        if args.dirs[0] == '':
-            args.dirs[0] = cephfs.getcwd().decode('utf-8')
         for dir_ in args.dirs :
-            for i in reversed(sorted(set(dirwalk(dir_)))):
-                try :
-                    print('{:10s} {}'.format(humansize(int(cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes').decode('utf-8'))), '.' + i))
-                except :
-                    continue
+            print('{:10s} {}'.format(humansize(int(cephfs.getxattr(to_bytes(dir_), 'ceph.dir.rbytes').decode('utf-8'))), dir_))
       
     def do_help(self, line):
         """
````

Reply:
But du command summarizes the disk usage for a directory with all its sub-directories right? But here it only gives the total disk size of the directory?